src/Kerbero.Infrastructure/Clients/NukiHttpClient.cs

```diff
@@ -0,0 +1,70 @@
+using Flurl;
+using Flurl.Http;
+using Kerbero.Common.Entities;
+using Kerbero.Common.Exceptions;
+using Kerbero.Infrastructure.Options;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Kerbero.Infrastructure.Clients;
+
+public class NukiHttpClient
+{
+	private readonly NukiClientOptions _options;
+
+	public NukiHttpClient(IOptions<NukiClientOptions> options)
+	{
+		_options = options.Value;
+	}
+
+	/// <summary>
+	/// Build the Uri for the redirect call to Nuki for the code
+	/// </summary>
+	/// <param name="clientId"></param>
+	/// <returns />
+	public Uri AskCodeAndGetUriForRedirect(string clientId)
+	{
+		if (string.IsNullOrEmpty(clientId)) throw new InvalidClientIdException(clientId, 1);
```

Comment:
```suggestion
		if (string.IsNullOrEmpty(clientId)) throw new InvalidClientIdException(clientId, 1);
```
Make the enum public instead of using a numeric value here, if we change the enum order this code will create an inconsistent exception